pkg/kube/inject/testdata/inject/proxy-override.yaml

```diff
@@ -49,6 +49,9 @@ spec:
           securityContext:
             allowPrivilegeEscalation: true
             readOnlyRootFilesystem: false
+            # These should not be removed or ignored, should be honored
+            runAsUser: 1234
+            runAsGroup: 1234
```

Comment:
Not sure why should we remove user's setting

Reply:
Indeed, this particular change in the test file was to guarantee we are not changing user values. This PR only changes it in a specific case: OpenShift.